pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -535,7 +535,7 @@ func (b *BridgeBindMechanism) decorateConfig(domainIface api.Interface) error {
 	for i, iface := range ifaces {
 		if iface.Alias.GetName() == b.iface.Name {
 			ifaces[i].MTU = domainIface.MTU
-			ifaces[i].MAC = &api.MAC{MAC: b.dhcpConfig.MAC.String()}
+			ifaces[i].MAC = domainIface.MAC
```

Comment:
I am confused.
1. What is the difference between `domainIface` and `domainIface`?
2. Don't we ignore the case the [MAC is not defined in the spec](https://github.com/kubevirt/kubevirt/blob/24b8d6859dde32301ab8c1ff8f7feaf35e761f6e/pkg/virt-launcher/virtwrap/network/podinterface.go#L484) here?

Reply:
> I am confused.
> 
> 1. What is the difference between `domainIface` and `b.iface`?

`b.iface` is a `v1.Interface` - the interface defined in the kubevirt spec - while `domainIface` represents an interface from the domXML. Entirely different animals.

> 2. Don't we ignore the case the [MAC is not defined in the spec](https://github.com/kubevirt/kubevirt/blob/24b8d6859dde32301ab8c1ff8f7feaf35e761f6e/pkg/virt-launcher/virtwrap/network/podinterface.go#L484) here?

If you check above the body of the function that generates this `domainIface`, you'll see it sets the MAC from `b.dhcpConfig.MAC.String`: https://github.com/kubevirt/kubevirt/blob/e5a1b7ae76e732d9ade0637b30ef3ccb25d2e16c/pkg/virt-launcher/virtwrap/network/podinterface.go#L524

which in turn is set from:
https://github.com/kubevirt/kubevirt/blob/e5a1b7ae76e732d9ade0637b30ef3ccb25d2e16c/pkg/virt-launcher/virtwrap/network/podinterface.go#L419

It's confusing, but has been like this since the dawn of time, and is addressed a few commits further along this PR, in https://github.com/kubevirt/kubevirt/pull/5607/commits/9aa22535c7d977612e2a2b34edc7f6d897247153 (the previous link will die once I have to force push...).

Reply:
I guess I was used to the previous mess.
So `b.iface` is the `vmSpecIface`.
`domainIface` is `extraLibvirtDomainIfaceParams`.
And `ifaces` is `libvirtDomainIfaces`.

I think we still have a work with the names (using my suggestions here or other).

Reply:
This PR is one of a series of PRs doing an overall refactoring the `phase1` and `phase2`. So if during the review I notice something related to the refactoring, I mention it in the PR. Do you have another place where I can add general comments regarding the refactoring?

Reply:
No, writing them here is ok; I'm compiling these into a wishlist (when I'm not opening small PRs that do it immediately).

I'm simply pointing this out to adjust your expectations of this PR. When you write a comment like this I'm aware you think something must be updated, but I don't understand in which PR.

Would updating the `notes for your reviewer` section of the PRs description be a good place to keep track of the list of open issues / track follow-up PRs ?

Reply:
Once the PR is merged I can create the whatever entity you want. 

Up till then, I'd rather not have to manage yet another document.

The jira with the epic has to be updated no matter what, never the less.